### DISCUSSION TITLE
Support_Time Warp View Modifications

### DISCUSSION THREAD

**Discussion by mkv929** [deejay-hub/timeline-lwc#239] [Q&A] [answered]
Hi Team,

Need some support and guidance on below requirements: - 

1. **For Task / Log A Call, can we**:
- Add Description, maybe three lines of text?
- Change the Due Date to Created Date?
- Add Last Modified By Name? 

2. For Content Note type can we remove the File Extension field?  
3. Same for the File type?  And could we replace with File Description?

_While removing the File Extension, I'm getting error._

Also, would be good to get visually the status of tasks, i.e. bold or change the color when completed.  Is this possible?

Thank You,
Manoj

**Comment by deejay-hub** [comment, marked as answer]
Hi @mkv929. Linking the config guide linked on the AppExchange listing for your reference (https://quip.com/ahFCA9VBEKtr). It might help explain how things can be adjusted and answer other questions you might have.

I'll try and answer you questions
1a. For Tasks you can change the **Fallback Tooltip Field** on the metadata record to show one field. We use Status by default. We don't suggest using Description since it's a large text area field and will cause issues for large data volumes. If you really want to see description you'll have to create a new custom field on Activity then copy the first x characters into it using Flow or Apex. Then use this custom field in the Fallback Tooltip Field. The following example is for the Account/Task relationship.

<img width="1170" alt="image" src="https://user-images.githubusercontent.com/7021883/223031618-6a7df85f-9d0a-47d9-8957-70fd43cd2ded.png">

1b. The date displayed is directly related to the date used to plot the Task on the Timeline. This comes from the Position Date Field in the metadata record. You can update it to CreatedDate. This will then plot the Task based on the date it was created and show you this value in the hover.

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/7021883/223031872-2255cfc8-68a6-4791-a9ae-b154d8541591.png">

1c. Time Warp uses the compact layout for the hover. Tasks are a special exception at Salesforce. Right now you're limited to one field to pick. However you can use a formula field for the Fallback Tooltip. That way you can have a formula that concatenates 2 or more field values together.

2. **You have to specify a fallback tooltip field** (which is why you get an error) but you can overwrite the value we currently use by default to be whatever you want. Just **avoid** large text area field data types.

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/7021883/223032586-dd49f35a-2004-4db5-bd3e-c178bd20d5a3.png">

3. See above. Notes and Files come from the same relationship. Any changes apply to both. You could in theory change the Fallback Tooltip to be ContentDocument.Description but it's not recommended since it's a large text area and causes issues for large data volumes. Since you can't add fields to this object your options are limited. The user can click the Note or File to see a preview with one click.

Visual status of tasks when complete isn't possible with the managed package. You'd have to add the code yourself. We have an open issue to add this functionality (https://github.com/deejay-hub/timeline-lwc/issues/27) but we're not actively working on it.

**Comment by mkv929** [comment]
Hi @deejay-hub , thank you so much for the quick review and reply. This is clear to me.  One more question - **Is it possible to show multiple fields with tooltip, somehow without creating any additional formula field?**

Thank You,
Manoj

**Reply by deejay-hub** [reply]
Any object in this list (https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_all_supported_objects.htm) can be controlled by the compact layout for the object. You will see however that Salesforce doesn't support ContentDocument or Activity or Task or Event. So for the objects in this discussion (Task and ContentDocument) we have resorted to a fallback - limited to 1 field. We don't currently provide a way to show more than 1 additional field. So formula fields is the best work around until Salesforce adds support for these objects.

**Reply by mkv929** [reply]
Thank you so much!
